pkg/monitoring/metrics/virt-controller/vmsnapshot.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ *
+ */
+
+package virt_controller
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	snapshotv1 "kubevirt.io/api/snapshot/v1beta1"
+)
+
+var (
+	vmSnapshotMetrics = []operatormetrics.Metric{
+		VmSnapshotSucceededTimestamp,
+	}
+
+	VmSnapshotSucceededTimestamp = operatormetrics.NewGaugeVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_vmsnapshot_succeeded_timestamp_seconds",
+			Help: "Returns the timestamp of successful virtual machine snapshot.",
+		},
+		[]string{"name", "snapshot_name", "namespace"},
+	)
+)
+
+func NewVMSnapshotCreated(snapshot *snapshotv1.VirtualMachineSnapshot) {
+	if snapshot.Status != nil && snapshot.Status.Phase == snapshotv1.Succeeded {
```

Comment:
You don't need to check for `snapshot.Status != nil` because the caller took care for this already. If you intend to use this function as generic handler I think its more readable to use a switch case here on the status phase

Reply:
sorry you are right it's not needed. removed